engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/schemaregistry/confluent/serialization/ConfluentKafkaAvroSerialization.scala

```diff
@@ -0,0 +1,16 @@
+package pl.touk.nussknacker.engine.avro.schemaregistry.confluent.serialization
+
+import org.apache.avro.Schema
+import org.apache.kafka.common.errors.SerializationException
+import pl.touk.nussknacker.engine.avro.AvroUtils
+import pl.touk.nussknacker.engine.avro.schemaregistry.confluent.client.ConfluentSchemaRegistryClient
+
+private[serialization] trait ConfluentKafkaAvroSerialization {
+
+  protected def schemaByTopicAndVersion(confluentSchemaRegistry: ConfluentSchemaRegistryClient, topic: String, version: Option[Int], isKey: Boolean): Schema = {
+    val subject = AvroUtils.topicSubject(topic, isKey = isKey)
+    confluentSchemaRegistry
+      .getFreshSchema(subject, version)
```

Comment:
W8, why you are fetching schema in each `deserialize` invocation? I've missed it because it was in mixed in class...